exchanges/gateio/gateio_test.go

```diff
@@ -74,838 +71,2829 @@ func TestStart(t *testing.T) {
+		t.Errorf("%s UpdateOrderbook() error %v", g.Name, err)
+	}
+}
+
+func TestGetWithdrawalsHistory(t *testing.T) {
+	t.Parallel()
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	if _, err := g.GetWithdrawalsHistory(context.Background(), currency.BTC, asset.Empty); err != nil {
+		t.Errorf("%s GetWithdrawalsHistory() error %v", g.Name, err)
+	}
+}
+func TestGetRecentTrades(t *testing.T) {
+	t.Parallel()
+	currencyPair := currency.NewPair(currency.BTC, currency.USDT)
+	_, err := g.GetRecentTrades(context.Background(), currencyPair, asset.Spot)
+	if err != nil {
+		t.Error(err)
+	}
+	cp, err := getFirstTradablePair(t, asset.DeliveryFutures)
+	if err != nil {
+		t.Fatal(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), cp, asset.DeliveryFutures)
+	if err != nil {
+		t.Error(err)
+	}
+	_, err = g.GetRecentTrades(context.Background(), currencyPair, asset.Options)
+	if err != nil {
+		t.Error(err)
+	}
+}
+
+func TestSubmitOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	var orderSubmission = &order.Submit{
+		Exchange: g.Name,
+		Pair: currency.Pair{
+			Delimiter: currency.UnderscoreDelimiter,
+			Base:      currency.BTC,
+			Quote:     currency.USDT,
+		},
+		Side:      order.Buy,
+		Type:      order.Limit,
+		Price:     1,
+		Amount:    1,
+		ClientID:  "meowOrder",
+		AssetType: asset.CrossMargin,
+	}
+	_, err := g.SubmitOrder(context.Background(), orderSubmission)
+	if err != nil {
+		t.Errorf("Order failed to be placed: %v", err)
+	}
+}
+
+func TestCancelExchangeOrder(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	var orderCancellation = &order.Cancel{
+		OrderID:       "1",
+		WalletAddress: core.BitcoinDonationAddress,
+		AccountID:     "1",
+		Pair:          currency.NewPair(currency.LTC, currency.BTC),
+		AssetType:     asset.Spot,
+	}
+	err := g.CancelOrder(context.Background(), orderCancellation)
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestCancelBatchOrders(t *testing.T) {
+	if !areTestAPIKeysSet() || !canManipulateRealOrders {
+		t.Skip()
+	}
+	currencyPair := currency.NewPair(currency.LTC, currency.BTC)
+	_, err := g.CancelBatchOrders(context.Background(), []order.Cancel{
+		{
+			OrderID:       "1",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}, {
+			OrderID:       "2",
+			WalletAddress: core.BitcoinDonationAddress,
+			AccountID:     "1",
+			Pair:          currencyPair,
+			AssetType:     asset.Spot,
+		}})
+	if err != nil && !strings.Contains(err.Error(), "ORDER_NOT_FOUND") {
+		t.Errorf("%s CancelOrder error: %v", g.Name, err)
+	}
+}
+
+func TestGetDepositAddress(t *testing.T) {
+	if !areTestAPIKeysSet() {
+		t.SkipNow()
+	}
+	_, err := g.GetDepositAddress(context.Background(), currency.USDT, "", "TRX")
```

Comment:
It seems like all authenticated endpoints are having issues, please go over all endpoints again and fix issues:

```
GateIO attempt 1 request path: https://api.gateio.ws/api/v4//api/v4//wallet/deposit_address?currency=USDT
```